pkg/proctree/proctree.go

```diff
@@ -0,0 +1,177 @@
+package proctree
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/logger"
+)
+
+//
+// The way the process tree is organized:
+//
+// 1. All processes and thread group leaders are stored as "processes" (hash -> process).
+// 2. All threads are stored as "threads" (hash -> thread).
+// 3. Each process has a list of children (hashes of real forked processes).
+// 4. Each process, that is a thread group leader, has a list of threads (hashes of threads).
+// 5. Each thread has a parent (hash of parent) and a leader (hash of thread group leader).
+// 6. Threads from the same thread group have the same parent, the parent of the group leader.
+//
+// Instruction:
+//
+// 1. To add an artifact to a process, simply add it to the process entry using its hash.
+// 2. To add an artifact to a process parent, use the process parent hash.
+// 3. To add an artifact to a thread, pick the group leader hash, add it to its process entry.
+//
+// NOTE: The importance of having the thread group leader hash to each thread is the following: the
+// thread group leader is considered a "Process", in the proctree, and it will have all the
+// information regarding the process, such as the executable and interpreter. Whenever an artifact
+// log, for a thread, is needed, the thread group leader hash will be used to find the process so it
+// can be logged as a process artifact.
+//
+
+const (
+	DefaultProcessCacheSize = 32768
+	DefaultThreadCacheSize  = 32768
+)
+
+type ProcTreeConfig struct {
+	Enabled          bool
```

Comment:
The `Enabled` flag is not relevant for the `ProcessTree` struct.
You should divide the config to one for the process tree initialization and the other is for user configuration.

Reply:
But all 3 are set by the user (cmdline) and, despite that, the Enabled is temporary (proctree will be always enabled soon). I'll leave as is for now.